src/crypto/sha3.cpp

```
@@ -0,0 +1,139 @@
+    0x000000000000800aULL, 0x800000008000000aULL, 0x8000000080008081ULL,
+    0x8000000000008080ULL, 0x0000000080000001ULL, 0x8000000080008008ULL
+};
+
+// Rotation offsets
+static const int keccak_rotation_offsets[25] = {
+    0, 1, 62, 28, 27, 36, 44, 6, 55, 20, 3, 10, 43,
+    25, 39, 41, 45, 15, 21, 8, 18, 2, 61, 56, 14
+};
+
+static inline uint64_t rotl64(uint64_t x, int n) {
+    return (x << n) | (x >> (64 - n));
```

Cppcheck (2.19.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)